test/benchmark.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh
+
+_SUBSTRUCTION () {
+    echo | awk "{print $1-$2}"
+}
+
+LOG_TABLE () {
+    time_begin=$(date +%s -d "$(awk '/.*DEBUG Hexo version/{print $1}' build.log)")
+    time_process_start=$(date +%s.%3N -d "$(awk '/.*INFO  Start processing/{print $1}' build.log)")
+    time_render_start=$(date +%s.%3N -d "$(awk '/.*INFO  Files loaded in/{print $1}' build.log)")
+    time_render_finish=$(date +%s.%3N -d "$(awk '/.*INFO.*generated in /{print $1}' build.log)")
+    time_database_saved=$(date +%s.%3N -d "$(awk '/.*DEBUG Database saved/{print $1}' build.log)")
+
+    echo "Load Plugin/Scripts/Database $(_SUBSTRUCTION $time_process_start $time_begin)\nProcess Source $(_SUBSTRUCTION $time_render_start $time_process_start)\nRender Files $(_SUBSTRUCTION $time_render_finish $time_render_start)\nSave Database $(_SUBSTRUCTION $time_database_saved $time_render_finish)" | awk '{printf "| %-28s | %7.3fs |\n",$1" "$2,$3"s"}'
```

Comment:
IMHO this line too long. How about split like following?

```sh
_MESSAGE_FORMATTER () {
    awk '{printf "| %-28s | %7.3fs |\n",$1" "$2,$3"s"}'
}

LOG_TABLE () {
    time_begin=$(date +%s -d "$(awk '/.*DEBUG Hexo version/{print $1}' build.log)")

    ...

    load_message="Load Plugin/Scripts/Database $(_SUBSTRUCTION $time_process_start $time_begin)"
    process_source_message="Process Source $(_SUBSTRUCTION $time_render_start $time_process_start)"
    render_files_message="Render Files $(_SUBSTRUCTION $time_render_finish $time_render_start)"
    save_database_message="Save Database $(_SUBSTRUCTION $time_database_saved $time_render_finish)"

    echo $load_message | _MESSAGE_FORMATTER
    echo $process_source_message | _MESSAGE_FORMATTER
    echo $render_files_message | _MESSAGE_FORMATTER
    echo $save_database_message | _MESSAGE_FORMATTER
}
```

Reply:
Updated.